src/Autofac.Multitenant/MultitenantContainer.cs

```diff
@@ -79,13 +80,12 @@ public class MultitenantContainer : Disposable, IContainer
     /// is <see cref="object"/>, value is <see cref="ILifetimeScope"/>.
     /// </summary>
     // Issue #280: Incorrect double-checked-lock pattern usage in MultitenantContainer.GetTenantScope
-    private readonly Dictionary<object, ILifetimeScope> _tenantLifetimeScopes = new();
+    private readonly ConcurrentDictionary<object, ILifetimeScope> _tenantLifetimeScopes = new();
 
     /// <summary>
-    /// Reader-writer lock for locking modifications and initializations
-    /// of tenant scopes.
+    /// Flag that disallows creating a new scope while disposing or after that.
     /// </summary>
-    private readonly ReaderWriterLockSlim _readWriteLock = new();
+    private int _isDisposed = 0;
```

Comment:
```suggestion
    private int _isDisposed;
```

Analyzers not liking the explicit initialization to default value.

Reply:
Sry, fixed